regression/rt_ar_react_1x2v_quadraticV.c

```diff
@@ -0,0 +1,969 @@
+        }, 
+        { .react_id = GKYL_REACT_RECOMB,
+          .type_self = GKYL_SELF_ION,
+          .ion_id = GKYL_ION_AR,
+          .elc_nm = "elc",
+          .ion_nm = "Ar2",
+          .recvr_nm = "Ar1",
+          .charge_state = 1,
+          .ion_mass = ctx.massAr,
+          .elc_mass = ctx.massElc,
+        },
+      },
+      },
+    
+    .bcx = {
+      .lower = { .type = GKYL_SPECIES_ZERO_FLUX, },
+      .upper = { .type = GKYL_SPECIES_ZERO_FLUX, },
+    },
+
+    /*
+    .bcx = {
+       .lower = {.type = GKYL_SPECIES_GK_SHEATH, },
+       .upper = {.type = GKYL_SPECIES_GK_SHEATH, },
+    },
+    */
+    .num_diag_moments = 5,
+    .diag_moments = { "M0", "M1", "M2", "M2par", "M2perp" },
+  };
+  
+  // field
+  struct gkyl_gyrokinetic_field field = {
+    .fem_parbc = GKYL_FEM_PARPROJ_PERIODIC, 
+    .kperpSq = pow(ctx.kperp, 2.),
+  };
+
+  // GK app
+  struct gkyl_gk gk = {
+    .name = "ar_react_1x2v_quad_wJacobvel",
+
+    //.cfl_frac = 0.1,
+
+    .cdim = 1, .vdim = 2,
+    .lower = { -ctx.Lz/2.0 },
+    .upper = { ctx.Lz/2.0 },
+    .cells = { NX },
+    .poly_order = 1,
+    .basis_type = app_args.basis_type,
+
+    .geometry = {
+      .geometry_id = GKYL_MAPC2P,
+      .world = {0.0, 0.0},
+      .mapc2p = mapc2p, // mapping of computational to physical space
+      .c2p_ctx = &ctx,
+      .bmag_func = bmag_func, // mapping of computational to physical space
+      .bmag_ctx = &ctx
+    },
+
+    .num_periodic_dir = 1,
+    .periodic_dirs = {0},
+
+    .num_species = 4,
+    .species = { elc, ion, Ar1, Ar2},
+
+    .field = field,
+
+
+
+    .use_gpu = app_args.use_gpu,
+    .skip_field=true,
+  };
+  // create app object
+  gkyl_gyrokinetic_app *app = gkyl_gyrokinetic_app_new(&gk);
+
+  // start, end and initial time-step
+  double tcurr = 0.0, tend = ctx.finalTime;
+  double dt = tend-tcurr;
+  int nframe = ctx.numFrames;
+  // create trigger for IO
+  struct gkyl_tm_trigger io_trig = { .dt = tend/nframe };
+
+  // initialize simulation
+  gkyl_gyrokinetic_app_apply_ic(app, tcurr);
+  write_data(&io_trig, app, tcurr);
+  gkyl_gyrokinetic_app_calc_field_energy(app, tcurr);
+
+  long step = 1, num_steps = app_args.num_steps;
+  while ((tcurr < tend) && (step <= num_steps)) {
+    //gkyl_gyrokinetic_app_cout(app, stdout, "Taking time-step at t = %g ...", tcurr);
+    struct gkyl_update_status status = gkyl_gyrokinetic_update(app, dt);
+    //gkyl_gyrokinetic_app_cout(app, stdout, " dt = %g\n", status.dt_actual);
+    if (step % 1 == 0) {
+      gkyl_gyrokinetic_app_calc_field_energy(app, tcurr);
+      gkyl_gyrokinetic_app_cout(app, stdout, "Taking time-step at t = %g ...", tcurr);
+      gkyl_gyrokinetic_app_cout(app, stdout, " dt = %g\n", status.dt_actual);
+    }
+    if (!status.success) {
+      gkyl_gyrokinetic_app_cout(app, stdout, "** Update method failed! Aborting simulation ....\n");
+      break;
+    }
+    tcurr += status.dt_actual;
+    dt = status.dt_suggested;
+
+    write_data(&io_trig, app, tcurr);
```

Comment:
follow the same logic/structure in time setting, time stepping, writing data, etc, as established reg tests.

Reply:
I updated the regression test. To follow https://github.com/ammarhakim/gkylzero/pull/469 required a merge which seems to have introduced a merge conflict which @manauref should probably resolve because of its location.